cpplint.py

```diff
@@ -50,12 +50,23 @@
 import sre_compile
 import string
 import sys
+import itertools
 import unicodedata
 
+# Files with any of these extensions are considered to be
+# header files (and will undergo different style checks).
+# This set can be extended by using the --headers
+# option (also supported in CPPLINT.cfg)
+_header_extensions = set(['h', 'hpp', 'hxx', 'h++', 'cuh'])
+_nonheader_extensions = set(['c', 'cc', 'cpp', 'cxx', 'c++', 'cu'])
+
+
 # The allowed extensions for file names
 # This is set by --extensions flag.
-_valid_extensions = set(['c', 'cc', 'cpp', 'cxx', 'c++', 'h', 'hpp', 'hxx',
-    'h++'])
+_valid_extensions = _nonheader_extensions.union(_header_extensions)
```

Comment:
I agree that the confusion already exists, but I disagree that this does not make matters worse.

By separating the extensions to several lists, and exposing only one of them to user-accessible configuration, making it possible that the lists go out of sync - I think this makes things worse (in some aspects) compared to the current state.

Unless I'm missing something...

I'll also be happy with leaving a TODO to fix this in a followup PR, assuming it's coming up soon :-)

Another option is to "unexpose" the `extensions` option until this gets settled in a followup PR.